lib/msal-common/src/telemetry/performance/IPerformanceClient.ts

```diff
@@ -4,34 +4,33 @@
  */
 
 import {
-    Counters,
     PerformanceEvent,
-    PerformanceEvents,
-    StaticFields,
+    PerformanceEvents
 } from "./PerformanceEvent";
 import { IPerformanceMeasurement } from "./IPerformanceMeasurement";
 
 export type PerformanceCallbackFunction = (events: PerformanceEvent[]) => void;
 
 export type InProgressPerformanceEvent = {
-    endMeasurement: (
+    end: (
         event?: Partial<PerformanceEvent>
     ) => PerformanceEvent | null;
-    discardMeasurement: () => void;
-    addStaticFields: (staticFields: StaticFields) => void;
-    increment: (counters: Counters) => void;
+    discard: () => void;
+    add: (fields: { [key: string]: {} | undefined; }) => void;
+    increment: (fields: { [key: string]: number | undefined; }) => void;
     event: PerformanceEvent;
```

Comment:
Why undefined? Can we not determine the type in advance? Or is it to keep it generic?

Reply:
To keep it compatible with the existing data points that can be undefined. Let me check if i can fix that without making breaking changes.

Reply:
We have to keep `undefined` for now due to the handful of data points that can be null/undefined.